### PR TITLE
tests: add documentation for test automation #229

### LINKED ISSUE #229: Add user support to run mixture of automation test suites
### Problem

The repo has a good wealth of test automation for smart contratc scenarios.
Most of this is run in Github actions and it may not be clear to users how to run individual tests and the whole test suite in various combinations.

### Solution

Add full test suite documentation and command line run scenarios.

- [ ] Add a README under https://github.com/hashgraph/hedera-smart-contracts/tree/main/test explaining test layout 
- [ ] Add instructions to run an individual suite - these will likely mimic what github actions do
- [ ] Add instructions to run the whole suite in a simple manner that gives confidence of result
- [ ] Add instructions to point single and full suite test to remote environments i.e. mainnet, testnet and previewnet (not using localnode)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by attest08**
**Related issue(s)**:

Fixes #229 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by natanasow — DISMISSED** [review]

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        "disableOptimisticBPs": true,
+        "cwd": "${workspaceFolder}"
+      }
+    ]
+  }
+```
+
+4. Use the Play button from the Run & Debug section to execute the tests. 
+
+![launch.json](/../hedera-smart-contracts/launch.png)
```
Comment (Nana-EC) [inline test/README.md:53]: Move to subfolder `./test/images` for now
```suggestion
![launch.json](/../hedera-smart-contracts/test/images/launch.png)
```

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- SafeHTS
+
+# Hedera Smart Contracts - Local test suite execution
+The tests are implmented in an independent manner, so they can be executed all, by test set or a single test.
+
+## Using command line:
+Running "npx hardhat test" in the terminal will execute all the existing tests. If you want to execute specific test set or single test you should:
+1. Open the desired test suite - Ex. ERC20.js
+2. Add "only" to the desired describe/it - Ex. describe.only || it.only
+3. Run in the terminal "npx hardhat test"
```
Comment (Nana-EC) [inline test/README.md:19]: Q: is there not a way to filter tests by file through command line?
That would be better if possible than devs having to change code

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Hedera Smart Contracts - tests
```
Comment (Nana-EC) [inline test/README.md:1]: Add a link to this README in the root README

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice. Some suggestions

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**